#335-disabled-checkboxes-while-loading

vitakh opened the pull request:
## Changes

Added a disable feature for check boxes: they get disabled while loading. 

## Notes

I was not able to test it out on the front-end app because the loading page was too quick.

## Checklist

- [x] All commits are tagged with the ticket number
- [x] No linting errors / newline at end of file warnings
- [x] All code follows repository-configured prettier formatting
- [x] No merge conflicts
- [x] All checks passing
- [x] Assign the PR to yourself
- [x] No `yarn.lock` changes (unless dependencies have changed)
- [x] Request reviewers & ping on Slack
- [x] PR is linked to the ticket (fill in the closes line below)

Closes #335

callmejim1226 reviewed (commented):
Great work! Although, I misread your ticket when I tried to help you. It's my bad, sorry. 

So your ticket was referring to the work package's checklist instead of the project's risk checklist. In the frontend app, you could find the work package page by clicking any existing work package inside the project page. What you can do is you could move your current code to workpackageviewcontainer.tsx file and use isLoading to the 'deliverables' and 'expected activities' checklist's isDisabled field. 

Although I'm not 100% sure if this ticket is meant to replace the existing condition in both checklists' isDisabled field to isLoading. You can ask Reid or other lead members if you're not sure about it.

anthonybernardi commented:
to build off of Ji-min's comment, this is actually good idea for risks too so you can keep it here. Just instead of using the isLoading from the checkBullet hook, you can extract isLoading from the useEditRisk() thing too and use that

RChandler234 commented:
Not high priority, will be worked on again in the Spring by same dev

vitakh commented:
Added a disable feature for check boxes for work packages: they get disabled while loading. 
Added a disable feature for check boxes for risks and fixed some imports.

Was not able to test it due to the fast loading.

Peyton-McKee commented on src/frontend/src/pages/ProjectDetailPage/ProjectViewContainer/RiskLog.tsx:
```diff
@@ -53,6 +53,7 @@ const RiskLog: React.FC<RiskLogProps> = ({ projectId, wbsNum, projLead, projMana
   const [newDetail, setNewDetail] = useState('');
   const [show, setShow] = useState(false);
   const risksQuery = useGetRisksForProject(projectId);
+  const { isLoading } = useEditSingleRisk();
```
Instead of using useEditSingleRisk() again, you should put the isLoading before the mutate async in line 51 where you can see we are already calling useEditSingleRisk().

Peyton-McKee commented on src/frontend/src/pages/WorkPackageDetailPage/WorkPackageViewContainer/WorkPackageViewContainer.tsx:
```diff
@@ -43,6 +44,7 @@ const WorkPackageViewContainer: React.FC<WorkPackageViewContainerProps> = ({
   const [showStageGateModal, setShowStageGateModal] = useState<boolean>(false);
   const [anchorEl, setAnchorEl] = useState<null | HTMLElement>(null);
   const dropdownOpen = Boolean(anchorEl);
+  const { isLoading } = useCheckDescriptionBullet();
```
Instead of adding this react hook here, I think you should do something similar to my other comment in the checklist component where it calls useCheckDescriptionBullet() on line 38. Then when it disables the checkbox, add an or for the isLoading value there.

Peyton-McKee reviewed (changes requested):
Great work! this is definitely a little tricky with figuring out where to declare the isLoadings, lmk if you have any questions on the changes!

Peyton-McKee reviewed (approved):
LGTM!

anthonybernardi reviewed (approved):
awesome! nice work
